activerecord/lib/active_record/persistence.rb

```diff
@@ -55,6 +55,79 @@ def create!(attributes = nil, &block)
         end
       end
 
+      # Inserts a record (or multiple records) into the database. This method constructs a single SQL INSERT
+      # statement and sends it straight to the database. It does not instantiate the involved models and it does not
+      # trigger Active Record callbacks or validations. However, values passed to #insert_many will still go through
+      # Active Record's normal type casting and serialization.
+      #
+      # The +attributes+ parameter can be either a Hash or an Array of Hashes. These Hashes describe the
+      # attributes on the objects that are to be created.
+      #
+      # ==== Options
+      #
+      # [:returning]
+      #   (Postgres-only) An array of column names that should be returned for all successfully inserted rows.
+      #   By default, the primary key(s) will be returned for every inserted rows. Can be set to `false` to return nothing.
+      #
+      # [:on_conflict]
+      #   A hash that describes how the database should respond to duplicate rows:
+      #
+      #     { do: :nothing } — skip duplicate rows
+      #
+      #     { do: :update } — update duplicate rows (perform an UPSERT)
+      #
+      #   For databases that support it, the hash can also specify which of several unique indexes to handle conficts for:
+      #
+      #     { column: %w{ isbn }, do: :update }
+      #
+      #   For databases that support partial indexes, the hash can also identify the unique index with its conditional
+      #
+      #     { column: %w{ isbn }, where: "published_on IS NOT NULL", do: :update }
+      #
+      # ==== Examples
+      #
+      #   # Insert a single record
+      #   Book.insert_many title: 'Agile Web Development with Rails', author: 'David'
+      #
+      #   # Insert multiple records
+      #   Book.insert_many [
+      #     { title: 'Agile Web Development with Rails', author: 'David' },
+      #     { title: 'Eloquent Ruby', author: 'Russ Olsen' }
+      #   ]
+      #
+      #   # Insert multiple records and skip duplicates
+      #   Book.insert_many [
+      #     { id: 1, title: 'Agile Web Development with Rails', author: 'David' },
+      #     { id: 2, title: 'Eloquent Ruby', author: 'Russ Olsen' }
+      #   ],
+      #     on_conflict: { do: :nothing }
+      #
+      def insert_many(attributes, options = {})
```

Comment:
I think insert_multiple or bulk_insert would be a better method name since we may not be inserting many, just multiples.

Reply:
👍 Do you think `insert_all` makes sense? (I think the `_all` is slightly awkward, but it would match `update_all` and `delete_all` whose behavior it parallels.)

Reply:
I like `insert_all` for the parity 👍